diff -ruN old/lib/mm/memlock.c new/lib/mm/memlock.c
--- old/lib/mm/memlock.c	2020-03-26 11:26:37.000000000 +0000
+++ new/lib/mm/memlock.c	2020-06-12 17:57:46.230654829 +0000
@@ -221,7 +221,9 @@
 
 static void _release_memory(void)
 {
+#if 0
 	free(_malloc_mem);
+#endif
 }
 
 /*
diff -ruN old/tools/lvmcmdline.c new/tools/lvmcmdline.c
--- old/tools/lvmcmdline.c	2020-03-26 11:26:37.000000000 +0000
+++ new/tools/lvmcmdline.c	2020-06-12 17:42:44.476090689 +0000
@@ -3438,7 +3438,7 @@
 	 */
 	dm_set_name_mangling_mode(DM_STRING_MANGLING_NONE);
 
-	if (!(cmd = create_toolcontext(0, NULL, 1, 0,
+	if (!(cmd = create_toolcontext(0, NULL, 0, 0,
 			set_connections, set_filters))) {
 		udev_fin_library_context();
 		return_NULL;
