diff --git a/mkfs/proto.c b/mkfs/proto.c
index 1a7b3586..65d91b57 100644
--- a/mkfs/proto.c
+++ b/mkfs/proto.c
@@ -28,6 +28,8 @@ static void populate_from_dir(struct xfs_mount *mp, struct fsxattr *fsxp,
 static void walk_dir(struct xfs_mount *mp, struct xfs_inode *pip,
 		struct fsxattr *fsxp, char *path_buf, int path_len);
 static int preserve_atime;
+static uid_t force_uid;
+static gid_t force_gid;
 static int slashes_are_spaces;
 
 /*
@@ -985,9 +987,13 @@ parse_proto(
 	struct fsxattr		*fsx,
 	struct proto_source	*protosource,
 	int			proto_slashes_are_spaces,
-	int			proto_preserve_atime)
+	int			proto_preserve_atime,
+	uid_t 		proto_force_uid,
+	gid_t 		proto_force_gid)
 {
 	preserve_atime = proto_preserve_atime;
+	force_uid = proto_force_uid;
+	force_gid = proto_force_gid;
 	slashes_are_spaces = proto_slashes_are_spaces;
 
 	/*
@@ -1695,8 +1701,8 @@ handle_direntry(
 	}
 
 	struct cred creds = {
-		.cr_uid = file_stat.st_uid,
-		.cr_gid = file_stat.st_gid,
+		.cr_uid = force_uid != (uid_t)-1 ? force_uid : file_stat.st_uid,
+		.cr_gid = force_gid != (gid_t)-1 ? force_gid : file_stat.st_gid,
 	};
 
 	xname.name = (unsigned char *)entry->d_name;
@@ -1850,8 +1856,8 @@ populate_from_dir(
 	 * We first ensure we have the root inode.
 	 */
 	struct cred creds = {
-		.cr_uid = file_stat.st_uid,
-		.cr_gid = file_stat.st_gid,
+		.cr_uid = force_uid != (uid_t)-1 ? force_uid : file_stat.st_uid,
+		.cr_gid = force_gid != (gid_t)-1 ? force_gid : file_stat.st_gid,
 	};
 	mode = file_stat.st_mode;
 
diff --git a/mkfs/proto.h b/mkfs/proto.h
index 476f7851..c58d0e47 100644
--- a/mkfs/proto.h
+++ b/mkfs/proto.h
@@ -20,7 +20,9 @@ struct proto_source setup_proto(char *fname);
 void parse_proto(struct xfs_mount *mp, struct fsxattr *fsx,
 		 struct proto_source *protosource,
 		 int proto_slashes_are_spaces,
-		 int proto_preserve_atime);
+		 int proto_preserve_atime,
+		 uid_t proto_force_uid,
+		 gid_t proto_force_gid);
 void res_failed(int err);
 
 #endif /* MKFS_PROTO_H_ */
diff --git a/mkfs/xfs_mkfs.c b/mkfs/xfs_mkfs.c
index b3440772..fab05ffc 100644
--- a/mkfs/xfs_mkfs.c
+++ b/mkfs/xfs_mkfs.c
@@ -125,6 +125,8 @@ enum {
 enum {
 	P_FILE = 0,
 	P_ATIME,
+	P_UID,
+	P_GID,
 	P_SLASHES,
 	P_MAX_OPTS,
 };
@@ -725,6 +727,8 @@ static struct opt_params popts = {
 	.subopts = {
 		[P_FILE] = "file",
 		[P_ATIME] = "atime",
+		[P_UID] = "uid",
+		[P_GID] = "gid",
 		[P_SLASHES] = "slashes_are_spaces",
 		[P_MAX_OPTS] = NULL,
 	},
@@ -739,6 +743,20 @@ static struct opt_params popts = {
 		  .maxval = 1,
 		  .defaultval = 1,
 		},
+		{
+			.index = P_UID,
+			.conflicts = { { NULL, LAST_CONFLICT } },
+			.minval = 0,
+			.maxval = UINT_MAX,
+			.defaultval = SUBOPT_NEEDS_VAL,
+		},
+		{
+			.index = P_GID,
+			.conflicts = { { NULL, LAST_CONFLICT } },
+			.minval = 0,
+			.maxval = UINT_MAX,
+			.defaultval = SUBOPT_NEEDS_VAL,
+		},
 		{ .index = P_SLASHES,
 		  .conflicts = { { NULL, LAST_CONFLICT } },
 		  .minval = 0,
@@ -1098,6 +1116,8 @@ struct cli_params {
 	int	is_supported;
 	int	proto_slashes_are_spaces;
 	int	proto_atime;
+	uid_t proto_uid;
+	gid_t proto_gid;
 	int	data_concurrency;
 	int	log_concurrency;
 	int	rtvol_concurrency;
@@ -1227,7 +1247,7 @@ usage( void )
 /* naming */		[-n size=num,version=2|ci,ftype=0|1,parent=0|1]]\n\
 /* no-op info only */	[-N]\n\
 /* prototype file */	[-p fname]\n\
-/* populate from directory */	[-p dirname,atime=0|1]\n\
+/* populate from directory */	[-p dirname,atime=0|1,uid=n,gid=n]\n\
 /* quiet */		[-q]\n\
 /* realtime subvol */	[-r extsize=num,size=num,rtdev=xxx,rgcount=n,rgsize=n,\n\
 			    concurrency=num,zoned=0|1,start=n,reserved=n]\n\
@@ -2141,6 +2161,20 @@ proto_opts_parser(
 	case P_ATIME:
 		cli->proto_atime = getnum(value, opts, subopt);
 		break;
+	case P_UID:
+		long long uid = getnum(value, opts, subopt);
+		if (uid < 0 || uid > UINT32_MAX)
+			illegal_option(value, opts, subopt,
+				_("UID must be between 0 and 4294967295."));
+		cli->proto_uid = (uid_t)uid;
+		break;
+	case P_GID:
+		long long gid = getnum(value, opts, subopt);
+		if (gid < 0 || gid > UINT32_MAX)
+			illegal_option(value, opts, subopt,
+				_("GID must be between 0 and 4294967295."));
+		cli->proto_gid = (gid_t)gid;
+		break;
 	case P_FILE:
 		fallthrough;
 	default:
@@ -5948,6 +5982,8 @@ main(
 		.rtvol_concurrency = -1, /* auto detect non-mechanical rtdev */
 		.autofsck = FSPROP_AUTOFSCK_UNSET,
 		.imaxpct = -1, /* set sb_imax_pct automatically */
+		.proto_uid = (uid_t)-1,
+		.proto_gid = (gid_t)-1,
 	};
 	struct mkfs_params	cfg = {};
 
@@ -6272,7 +6308,9 @@ main(
 	parse_proto(mp, &cli.fsx,
 			&protosource,
 			cli.proto_slashes_are_spaces,
-			cli.proto_atime);
+			cli.proto_atime,
+			cli.proto_uid,
+			cli.proto_gid);
 
 	/*
 	 * Protect ourselves against possible stupidity
