--- include/configs/tegra210-common.h	2022-01-13 15:45:15.834014969 +0000
+++ include/configs/tegra210-common.h	2022-01-13 15:45:21.250066512 +0000
@@ -54,30 +54,10 @@
 #define MEM_LAYOUT_ENV_SETTINGS \
 	"scriptaddr=0x90000000\0" \
 	"pxefile_addr_r=0x90100000\0" \
-	"fdtoverlay_addr_r=0x90200000\0" \
 	"kernel_addr_r=" __stringify(CONFIG_LOADADDR) "\0" \
+	"fdtfile=" FDTFILE "\0" \
 	"fdt_addr_r=0x83000000\0" \
-	"ramdisk_addr_r=0x83200000\0" \
-	"fdt_copy_node_paths=" \
-		"/chosen/plugin-manager:" \
-		"/chosen/reset:" \
-		"/chosen/display-board:" \
-		"/chosen/proc-board:" \
-		"/chosen/pmu-board:" \
-		"/external-memory-controller@7001b000:" \
-		"/memory@80000000\0" \
-	"fdt_copy_prop_paths=" \
-		"/bpmp/carveout-start:" \
-		"/bpmp/carveout-size:" \
-		"/chosen/eks_info:" \
-		"/chosen/nvidia,bluetooth-mac:" \
-		"/chosen/nvidia,ethernet-mac:" \
-		"/chosen/nvidia,wifi-mac:" \
-		"/chosen/uuid:" \
-		"/chosen/linux,initrd-start:" \
-		"/chosen/linux,initrd-end:" \
-		"/serial-number:" \
-		"/psci/nvidia,system-lp0-disable\0"
+	"ramdisk_addr_r=0x83200000\0"
 
 /* For USB EHCI controller */
 #define CONFIG_EHCI_IS_TDI
