diff --git a/src/shared/selinux-util.c b/src/shared/selinux-util.c
index 84e45e3b7..1a3b1c1de 100644
--- a/src/shared/selinux-util.c
+++ b/src/shared/selinux-util.c
@@ -48,10 +48,10 @@ static struct selabel_handle *label_hnd = NULL;
 
 bool mac_selinux_use(void) {
 #ifdef HAVE_SELINUX
-        if (cached_use < 0)
-                cached_use = is_selinux_enabled() > 0;
+        // if (cached_use < 0)
+        //         cached_use = is_selinux_enabled() > 0;
 
-        return cached_use;
+        return true;
 #else
         return false;
 #endif
@@ -65,6 +65,8 @@ void mac_selinux_retest(void) {
 
 int mac_selinux_init(const char *prefix) {
         int r = 0;
+        printf("1 \n");
+        fflush(stdout);
 
 #ifdef HAVE_SELINUX
         usec_t before_timestamp, after_timestamp;
@@ -77,16 +79,22 @@ int mac_selinux_init(const char *prefix) {
 
         if (!mac_selinux_use())
                 return 0;
+        printf("11111 \n");
+        fflush(stdout);
 
         if (label_hnd)
                 return 0;
 
+printf("2 #ifdef HAVE_MALLINFO2\n");
+fflush(stdout);
 #ifdef HAVE_MALLINFO2
         before_mallinfo = mallinfo2();
 #elif HAVE_MALLINFO
         before_mallinfo = mallinfo();
 #endif
 
+printf("3         before_timestamp = now(CLOCK_MONOTONIC);\n");
+fflush(stdout);
         before_timestamp = now(CLOCK_MONOTONIC);
 
         if (prefix) {
@@ -98,24 +106,32 @@ int mac_selinux_init(const char *prefix) {
         } else
                 label_hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
 
+printf("4         if (!label_hnd) {\n");
+fflush(stdout);
         if (!label_hnd) {
                 log_enforcing("Failed to initialize SELinux context: %m");
                 r = security_getenforce() == 1 ? -errno : 0;
         } else  {
                 char timespan[FORMAT_TIMESPAN_MAX];
 
+printf("5 #if defined(HAVE_MALLINFO)|defined(HAVE_MALLINFO2)\n");
+fflush(stdout);
 #if defined(HAVE_MALLINFO)|defined(HAVE_MALLINFO2)
                 int l;
 #endif
 
                 after_timestamp = now(CLOCK_MONOTONIC);
 
+printf("6 #ifdef HAVE_MALLINFO2\n");
+fflush(stdout);
 #ifdef HAVE_MALLINFO2
                 after_mallinfo = mallinfo2();
 #elif HAVE_MALLINFO
                 after_mallinfo = mallinfo();
 #endif
 
+printf("7 #if defined(HAVE_MALLINFO)|defined(HAVE_MALLINFO2)\n");
+fflush(stdout);
 #if defined(HAVE_MALLINFO)|defined(HAVE_MALLINFO2)
                 l = after_mallinfo.uordblks > before_mallinfo.uordblks ? after_mallinfo.uordblks - before_mallinfo.uordblks : 0;
 
@@ -126,6 +142,8 @@ int mac_selinux_init(const char *prefix) {
                 log_debug("Successfully loaded SELinux database in %s",
                           format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0));
 #endif
+printf("8         }\n");
+fflush(stdout);
         }
 #endif
 
@@ -148,12 +166,18 @@ int mac_selinux_fix(const char *path, bool ignore_enoent, bool ignore_erofs) {
 #ifdef HAVE_SELINUX
         struct stat st;
         int r;
+        printf("9 \n");
+        fflush(stdout);
 
         assert(path);
+        printf("jj\n");
+        fflush(stdout);
 
         /* if mac_selinux_init() wasn't called before we are a NOOP */
         if (!label_hnd)
                 return 0;
+        printf("10 \n");
+        fflush(stdout);
 
         r = lstat(path, &st);
         if (r >= 0) {
@@ -173,6 +197,8 @@ int mac_selinux_fix(const char *path, bool ignore_enoent, bool ignore_erofs) {
                                 return 0;
                 }
         }
+        printf("11 \n");
+        fflush(stdout);
 
         if (r < 0) {
                 /* Ignore ENOENT in some cases */
@@ -186,6 +212,8 @@ int mac_selinux_fix(const char *path, bool ignore_enoent, bool ignore_erofs) {
                 if (security_getenforce() == 1)
                         return -errno;
         }
+printf("12 \n");
+fflush(stdout);
 #endif
 
         return 0;
@@ -196,17 +224,26 @@ int mac_selinux_apply(const char *path, const char *label) {
 #ifdef HAVE_SELINUX
         assert(path);
         assert(label);
+        printf("13 \n");
+        fflush(stdout);
 
-        if (!mac_selinux_use())
-                return 0;
+        if (!mac_selinux_use()) {
+                errno = EROFS;
+                return -EROFS;
+        }
+        printf("14 \n");
+        fflush(stdout);
 
         if (setfilecon(path, label) < 0) {
                 log_enforcing("Failed to set SELinux security context %s on path %s: %m", label, path);
                 if (security_getenforce() == 1)
                         return -errno;
         }
+        printf("15 \n");
+        fflush(stdout);
 #endif
-        return 0;
+        errno = ENOMEM;
+        return -ENOMEM;
 }
 
 int mac_selinux_get_create_label_from_exe(const char *exe, char **label) {
@@ -215,9 +252,13 @@ int mac_selinux_get_create_label_from_exe(const char *exe, char **label) {
 #ifdef HAVE_SELINUX
         _cleanup_security_context_free_ char *mycon = NULL, *fcon = NULL;
         security_class_t sclass;
+        printf("hh\n");
+        fflush(stdout);
 
         assert(exe);
         assert(label);
+        printf("16 \n");
+        fflush(stdout);
 
         if (!mac_selinux_use())
                 return -EOPNOTSUPP;
@@ -227,13 +268,19 @@ int mac_selinux_get_create_label_from_exe(const char *exe, char **label) {
                 return -errno;
 
         r = getfilecon(exe, &fcon);
+        printf("17 if (r < 0)\n");
+        fflush(stdout);
         if (r < 0)
                 return -errno;
 
         sclass = string_to_security_class("process");
+        printf("18 r = security_compute_create(mycon, fcon, sclass, label);\n");
+        fflush(stdout);
         r = security_compute_create(mycon, fcon, sclass, label);
         if (r < 0)
                 return -errno;
+        printf("19\n");
+        fflush(stdout);
 #endif
 
         return r;
@@ -245,12 +292,18 @@ int mac_selinux_get_our_label(char **label) {
         assert(label);
 
 #ifdef HAVE_SELINUX
+printf("20         if (!mac_selinux_use())\n");
+fflush(stdout);
         if (!mac_selinux_use())
                 return -EOPNOTSUPP;
+        printf("21 \n");
+        fflush(stdout);
 
         r = getcon(label);
         if (r < 0)
                 return -errno;
+        printf("22 \n");
+        fflush(stdout);
 #endif
 
         return r;
@@ -264,6 +317,8 @@ int mac_selinux_get_child_mls_label(int socket_fd, const char *exe, const char *
         _cleanup_context_free_ context_t pcon = NULL, bcon = NULL;
         security_class_t sclass;
         const char *range = NULL;
+        printf("aa\n");
+        fflush(stdout);
 
         assert(socket_fd >= 0);
         assert(exe);
@@ -271,6 +326,8 @@ int mac_selinux_get_child_mls_label(int socket_fd, const char *exe, const char *
 
         if (!mac_selinux_use())
                 return -EOPNOTSUPP;
+        printf("bb\n");
+        fflush(stdout);
 
         r = getcon(&mycon);
         if (r < 0)
@@ -299,6 +356,8 @@ int mac_selinux_get_child_mls_label(int socket_fd, const char *exe, const char *
         range = context_range_get(pcon);
         if (!range)
                 return -errno;
+        printf("22\n");
+        fflush(stdout);
 
         r = context_range_set(bcon, range);
         if (r)
@@ -313,6 +372,8 @@ int mac_selinux_get_child_mls_label(int socket_fd, const char *exe, const char *
         r = security_compute_create(mycon, fcon, sclass, label);
         if (r < 0)
                 return -errno;
+        printf("23\n");
+        fflush(stdout);
 #endif
 
         return r;
@@ -323,8 +384,12 @@ void mac_selinux_free(char *label) {
 #ifdef HAVE_SELINUX
         if (!mac_selinux_use())
                 return;
+        printf("24 \n");
+        fflush(stdout);
 
         freecon(label);
+        printf("25 \n");
+        fflush(stdout);
 #endif
 }
 
@@ -333,11 +398,17 @@ int mac_selinux_create_file_prepare(const char *path, mode_t mode) {
 
 #ifdef HAVE_SELINUX
         _cleanup_security_context_free_ char *filecon = NULL;
+        printf("26 \n");
+        fflush(stdout);
 
         assert(path);
+        printf("27 \n");
+        fflush(stdout);
 
         if (!label_hnd)
                 return 0;
+        printf("cc\n");
+        fflush(stdout);
 
         if (path_is_absolute(path))
                 r = selabel_lookup_raw(label_hnd, &filecon, path, mode);
@@ -350,10 +421,14 @@ int mac_selinux_create_file_prepare(const char *path, mode_t mode) {
 
                 r = selabel_lookup_raw(label_hnd, &filecon, newpath, mode);
         }
+        printf("dd\n");
+        fflush(stdout);
 
         /* No context specified by the policy? Proceed without setting it. */
         if (r < 0 && errno == ENOENT)
                 return 0;
+        printf("28 \n");
+        fflush(stdout);
 
         if (r < 0)
                 r = -errno;
@@ -365,8 +440,12 @@ int mac_selinux_create_file_prepare(const char *path, mode_t mode) {
                 }
         }
 
+printf("29         if (r < 0 && security_getenforce() == 0)\n");
+fflush(stdout);
         if (r < 0 && security_getenforce() == 0)
                 r = 0;
+        printf("30 \n");
+        fflush(stdout);
 #endif
 
         return r;
@@ -376,21 +455,33 @@ void mac_selinux_create_file_clear(void) {
 
 #ifdef HAVE_SELINUX
         PROTECT_ERRNO;
+        printf("31 \n");
+        fflush(stdout);
 
         if (!mac_selinux_use())
                 return;
+        printf("32\n");
+        fflush(stdout);
 
         setfscreatecon(NULL);
+        printf("33\n");
+        fflush(stdout);
 #endif
 }
 
 int mac_selinux_create_socket_prepare(const char *label) {
 
+printf("33 #ifdef HAVE_SELINUX\n");
+fflush(stdout);
 #ifdef HAVE_SELINUX
         if (!mac_selinux_use())
                 return 0;
+        printf("ee\n");
+        fflush(stdout);
 
         assert(label);
+        printf("ff\n");
+        fflush(stdout);
 
         if (setsockcreatecon(label) < 0) {
                 log_enforcing("Failed to set SELinux security context %s for sockets: %m", label);
@@ -398,6 +489,8 @@ int mac_selinux_create_socket_prepare(const char *label) {
                 if (security_getenforce() == 1)
                         return -errno;
         }
+        printf("34\n");
+        fflush(stdout);
 #endif
 
         return 0;
@@ -407,11 +500,17 @@ void mac_selinux_create_socket_clear(void) {
 
 #ifdef HAVE_SELINUX
         PROTECT_ERRNO;
+        printf("gg\n");
+        fflush(stdout);
 
         if (!mac_selinux_use())
                 return;
+        printf("35 \n");
+        fflush(stdout);
 
         setsockcreatecon(NULL);
+printf("36\n");
+fflush(stdout);
 #endif
 }
 
